framework/src/utils/Units.C

```diff
@@ -0,0 +1,541 @@
+    {"C", {1, {{MooseUnits::BaseUnit::AMPERE, 1}, {MooseUnits::BaseUnit::SECOND, 1}}}}, // Coulomb
+    {"V",
+     {
+         1,
+         {{MooseUnits::BaseUnit::KILOGRAM, 1},
+          {MooseUnits::BaseUnit::METER, 2},
+          {MooseUnits::BaseUnit::SECOND, -3},
+          {MooseUnits::BaseUnit::AMPERE, -1}},
+     }},
+    {"F",
+     {1,
+      {{MooseUnits::BaseUnit::KILOGRAM, -1},
+       {MooseUnits::BaseUnit::METER, -2},
+       {MooseUnits::BaseUnit::SECOND, 4},
+       {MooseUnits::BaseUnit::AMPERE, 2}}}},
+    {"S",
+     {1,
+      {{MooseUnits::BaseUnit::KILOGRAM, -1},
+       {MooseUnits::BaseUnit::METER, -2},
+       {MooseUnits::BaseUnit::SECOND, 3},
+       {MooseUnits::BaseUnit::AMPERE, 2}}}}, // Siemens = 1/Ohm (electrical conductance)
+    {"Wb",
+     {1,
+      {{MooseUnits::BaseUnit::KILOGRAM, 1},
+       {MooseUnits::BaseUnit::METER, 2},
+       {MooseUnits::BaseUnit::SECOND, -2},
+       {MooseUnits::BaseUnit::AMPERE, -1}}}}, // Weber (magnetic flux)
+    {"T",
+     {1,
+      {{MooseUnits::BaseUnit::KILOGRAM, 1},
+       {MooseUnits::BaseUnit::SECOND, -2},
+       {MooseUnits::BaseUnit::AMPERE, -1}}}}, // Tesla (magnetic flux density)
+    {"H",
+     {1,
+      {{MooseUnits::BaseUnit::KILOGRAM, 1},
+       {MooseUnits::BaseUnit::METER, 2},
+       {MooseUnits::BaseUnit::SECOND, -2},
+       {MooseUnits::BaseUnit::AMPERE, -2}}}}, // Henry (inductance)
+
+    // cgs units
+    {"Ba",
+     {0.1,
+      {{MooseUnits::BaseUnit::KILOGRAM, 1},
+       {MooseUnits::BaseUnit::METER, -1},
+       {MooseUnits::BaseUnit::SECOND,
+        -2}}}}, // barye (unit of pressure - not to be confused with bar)
+    {"dyn",
+     {1e-5,
+      {{MooseUnits::BaseUnit::KILOGRAM, 1},
+       {MooseUnits::BaseUnit::METER, 1},
+       {MooseUnits::BaseUnit::SECOND, -2}}}}, // dyne
+
+    // Freedom units
+    {"ft", {0.3048, {{MooseUnits::BaseUnit::METER, 1}}}},        // foot
+    {"in", {25.4e-3, {{MooseUnits::BaseUnit::METER, 1}}}},       // inch
+    {"lb", {0.45359237, {{MooseUnits::BaseUnit::KILOGRAM, 1}}}}, // pound of mass
+    {"lbf",
+     {4.4482216152605,
+      {{MooseUnits::BaseUnit::KILOGRAM, 1},
+       {MooseUnits::BaseUnit::METER, 1},
+       {MooseUnits::BaseUnit::SECOND, -2}}}}, // pound of force
+    {"psi",
+     {6.894757e3,
+      {{MooseUnits::BaseUnit::KILOGRAM, 1},
+       {MooseUnits::BaseUnit::METER, -1},
+       {MooseUnits::BaseUnit::SECOND, -2}}}}, // pound-force per square inch
+
+    // misc.
+    {"BTU",
+     {1055.06,
+      {{MooseUnits::BaseUnit::KILOGRAM, 1},
+       {MooseUnits::BaseUnit::METER, 2},
+       {MooseUnits::BaseUnit::SECOND, -2}}}}, // ISO 31-4 British thermal unit
+    {"bar",
+     {1e5,
+      {{MooseUnits::BaseUnit::KILOGRAM, 1},
+       {MooseUnits::BaseUnit::METER, -1},
+       {MooseUnits::BaseUnit::SECOND, -2}}}},                          // bar (unit of pressure)
+    {"h", {60 * 60, {{MooseUnits::BaseUnit::SECOND, 1}}}},             // day
```

Comment:
probably 'hour', rather than 'day'

Reply:
Also, why is there no 'minute' unit?

Reply:
Yeah... because that would create an ambiguity between milli-inch (min) and minute (min).
We could just accept that (as the ambiguous units have different dimensions - length vs. time), but I chose to avoid that for now.

There are potential paths for providing a disambiguation (by permitting an optional separating character between SI prefix and unit, making milli inch `m_in` or `m.in`), but I was reluctant to introduce that complication (at least in the first iteration).

Reply:
Why could not you do `min` for minute? Just asking...

Reply:
See above. min could be interpreted as milli inch.

Reply:
Got it. I guess I need to wake up first. Thanks!